Sources/NIOCore/AsyncSequences/NIOAsyncWriter.swift

```diff
@@ -293,7 +293,7 @@ public struct NIOAsyncWriter<
     /// - Parameter element: The element to yield.
     @inlinable
     public func yield(_ element: Element) async throws {
-        try await self._storage.yield(contentsOf: CollectionOfOne(element))
+        try await self._storage.yield(element: element)
```

Comment:
The fix here is quite simple. The problem was that we lost the fast path for single element writes in some refactoring. This PR duplicates the write path and specialises one of them for single element writes to call the optimised delegate method of the writer.